WorkflowAdminBundle/Controller/Admin/ParametersController.php

```diff
@@ -0,0 +1,141 @@
+        $data = array(
+            'statuses' => $statusCollection,
+            'labels'   => array(
+                $this->get('translator')
+                    ->trans('open_orchestra_workflow_admin.status.initial_state'          , array(), 'interface'),
+                $this->get('translator')
+                    ->trans('open_orchestra_workflow_admin.status.translation_state'      , array(), 'interface'),
+                $this->get('translator')
+                    ->trans('open_orchestra_workflow_admin.status.published_state'        , array(), 'interface'),
+                $this->get('translator')
+                    ->trans('open_orchestra_workflow_admin.status.auto_publish_from_state', array(), 'interface'),
+                $this->get('translator')
+                    ->trans('open_orchestra_workflow_admin.status.auto_unpublish_to_state', array(), 'interface'),
+            )
+        );
+
+        $form = $this->createForm('oo_workflow_parameters', $data, array(
+            'action' => $this->generateUrl('open_orchestra_workflow_admin_parameters_form')
+        ));
+
+        $updatedStatusIndexes = array();
+        if ($request->isMethod('POST')) {
+            $updatedStatusIndexes = $this->getStatusIndexesToUpdate(
+                $statusCollection,
+                $request->get('oo_workflow_parameters')
+            );
+        }
+
+        $form->handleRequest($request);
+        $message = $this->get('translator')->trans('open_orchestra_workflow_admin.form.parameters.success');
+        if ($this->handleForm($form, $message)) {
+            foreach ($updatedStatusIndexes as $statusIndex) {
+                $this->dispatchEvent(
+                    StatusEvents::STATUS_UPDATE,
+                    new StatusEvent($statusCollection[$statusIndex])
+                );
+            }
+        }
+
+        return $this->renderAdminForm($form);
+    }
+
+    /**
+     * Get the list of status to update in $statusCollection
+     * Only indexes in $statusCollection are returned
+     *
+     * @param array $statusCollection
+     * @param array $formData
+     *
+     * @return array
+     */
+    protected function getStatusIndexesToUpdate(array $statusCollection, array $formData)
+    {
+        $statusMap = array();
+        $updatedStatusIndexes = array();
+        $formStatuses = $formData['statuses'];
+
+        foreach ($statusCollection as $index => $status) {
+            $statusMap[$index] = array(
+                'initialState'         => $status->isInitialState(),
+                'translationState'     => $status->isTranslationState(),
+                'publishedState'       => $status->isPublishedState(),
+                'autoPublishFromState' => $status->isAutoPublishFromState(),
+                'autoUnpublishToState' => $status->isAutoUnpublishToState(),
+            );
+        }
+
+        foreach ($statusMap as $statusIndex => $oldParameters) {
```

Comment:
Could be done by :
foreach ($statusMap as $statusIndex => $oldParameters) {
    if (count(array_diff_assoc($oldParameters, $formStatuses[$statusIndex])) > 0) {
        $updatedStatusIndexes[] = $statusIndex;
    }
}